environment/src/main/java/com/sequenceiq/environment/environment/service/DistroxService.java

```diff
@@ -0,0 +1,82 @@
+package com.sequenceiq.environment.environment.service;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.concurrent.TimeUnit;
+
+import javax.ws.rs.ProcessingException;
+import javax.ws.rs.WebApplicationException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import com.dyngr.Polling;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.response.StackViewV4Response;
+import com.sequenceiq.distrox.api.v1.distrox.endpoint.DistroXV1Endpoint;
+
+@Service
+public class DistroxService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DistroxService.class);
+
+    @Value("${env.stop.polling.attempt:90}")
+    private Integer attempt;
+
+    @Value("${env.stop.polling.sleep.time:5}")
+    private Integer sleeptime;
+
+    private final DistroXV1Endpoint distroXV1Endpoint;
+
+    private final DistroxPollerCollection distroxPollerCollection;
+
+    public DistroxService(DistroXV1Endpoint distroXV1Endpoint, DistroxPollerCollection distroxPollerCollection) {
+        this.distroXV1Endpoint = distroXV1Endpoint;
+        this.distroxPollerCollection = distroxPollerCollection;
+    }
+
+    public Collection<StackViewV4Response> getAttachedDistroXClusters(String environmentName, String environmentCrn) {
+        LOGGER.debug("Get DistroX clusters of the environment: '{}'", environmentName);
+        try {
+            return distroXV1Endpoint.list(environmentName, environmentCrn).getResponses();
+        } catch (WebApplicationException | ProcessingException e) {
+            String message = String.format("Failed to get DistroX clusters from service due to: '%s' ", e.getMessage());
+            LOGGER.error(message, e);
+            throw e;
+        }
+    }
+
+    public void stopAttachedDistrox(String envName) {
+        Collection<StackViewV4Response> attachedDistroXClusters = getAttachedDistroXClusters(envName, null);
+        ArrayList<String> pollingCrn = new ArrayList<>();
+        attachedDistroXClusters.forEach(d -> {
+            if (!d.getStatus().isStopState()) {
+                distroXV1Endpoint.putStopByCrn(d.getCrn());
```

Comment:
If any of these rest calls fail for any reason, the whole stop operation will fail. There could be temporary network issues or anything.

There are 2 ways to mitigate this:
- better: introduce multi-stop and multi-start endpoints
- worse: handle various exceptions here